sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rule/BeamAggregationRule.java

```diff
@@ -181,12 +181,12 @@ private static RelNode updateWindow(RelOptRuleCall call, Aggregate aggregate, Pr
   }
 
   private static Duration durationParameter(List<RexNode> parameters, int parameterIndex) {
-    return Duration.millis(intValue(parameters.get(parameterIndex)));
+    return Duration.millis(longValue(parameters.get(parameterIndex)));
   }
 
-  private static long intValue(RexNode operand) {
+  private static long longValue(RexNode operand) {
     if (operand instanceof RexLiteral) {
-      return RexLiteral.intValue(operand);
+      return ((Number) RexLiteral.value(operand)).longValue();
```

Comment:
I also bet that adding this to Calcite at https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rex/RexLiteral.java#L1093 would be welcome.

Reply:
@kennknowles filed CALCITE-2837 with patch